README.md

```diff
@@ -57,48 +60,253 @@ This is work in progress as of 14 Nov 2018:
+| `method`             |                    | (**required**) [HTTP method](#suppported-http-methods)    |
+| `path`               |                    | (**required**) URL path                                   |
+| `description`        | `""`               | Description of the endpoint                               |
+| `requestContentType` | `application/json` | Content type of the request body                          |
+| `successStatusCode`  | `200`              | HTTP status code for a successful response                |
+| `tags`               |                    | Array of tags used for endpoint grouping in documentation |
+
+### `@request`
+
+Define a request body for requests that require one. This is commonly used for `POST` and `PUT` requests:
+
+```TypeScript
+class MyUserEndpoints {
+  //...
+  @endpoint({
+    method: "POST",
+    path: "/users"
+  })
+  createUser(
+    @request req: CreateUserRequest
+  ): UserResponse {
+    return response();
+  }
+
+  @endpoint({
+    method: "PUT",
+    path: "/users/:id"
+  })
+  updateUser(
+    @pathParam({ description: "User unique identifier" }) id: string,
+    @request req: UpdateUserRequest
+  ): UserResponse {
+    return response();
+  }
+  //...
+}
+
+interface CreateUserRequest {
+  firstName: string;
+  lastName: string;
+}
+
+interface UpdateUserRequest {
+  lastName: string;
+}
+```
+
+### `@header`
+
+Define a request header. `@header` can be used multiple times to define multiple headers.
+
+| Field         | Description                       |
+| ------------- | --------------------------------- |
+| `name`        | (**required**) Name of the header |
+| `description` | Description of the header         |
+
+### `@pathParam`
+
+Define path parameters that appear in the `@endpoint.path`. The parameter must be defined in the `@endpoint.path`. A `@pathParam` should be defined for each parameter defined in `@endpoint.path`.
+
+| Field         | Description                       |
+| ------------- | --------------------------------- |
+| `description` | Description of the path parameter |
+
+### `@queryParam`
+
+Define query parameters. `@queryParam` can be used multiple times to define multiple query parameters.
+
+| Field         | Description                        |
+| ------------- | ---------------------------------- |
+| `description` | Description of the query parameter |
+
+### `@specificError<T>`
+
+Define a known error for the endpoint. `@specificError` can be used multiple times to define multiple errors.
+
+| Field        | Description                                   |
+| ------------ | --------------------------------------------- |
+| `name`       | (**required**) Name of the error              |
+| `statusCode` | (**required**) HTTP status code for the error |
+
+### `@genericError<T>`
+
+Define a default error for the endpoint. This can only be used once for an `@endpoint`.
```

Comment:
```suggestion
Define a default error for the endpoint. This can only be used once for an `@endpoint`.

`T` must be replaced with the response type when the error occurs, for example `@specificError<UnauthenticatedErrorResponse>`.
```